nimare/dataset/dataset.py

```diff
@@ -88,3 +222,47 @@ def get_images(self):
 
     def get_coordinates(self):
         pass
+
+    def save(self, filename):
+        """
+        Pickle the Dataset instance to the provided file.
+        If the filename ends with 'z', gzip will be used to write out a
+        compressed file. Otherwise, an uncompressed file will be created.
+        """
+        if filename.endswith('z'):
+            with gzip.GzipFile(filename, 'wb') as file_object:
+                pickle.dump(self, file_object)
+        else:
+            with open(filename, 'wb') as file_object:
+                pickle.dump(self, file_object)
+
+    @classmethod
+    def load(cls, filename):
```

Comment:
Addressing concerns raised in #2, I replaced the filename-based check with a `compressed` argument in my new PR (#5).